pkgs/development/tools/systemfd/default.nix

```diff
@@ -0,0 +1,18 @@
+{ stdenv, lib, fetchgit, darwin, buildPlatform
+, buildRustCrate, buildRustCrateHelpers, defaultCrateOverrides }:
+
+((import ./Cargo.nix {
+  inherit lib buildPlatform buildRustCrate buildRustCrateHelpers fetchgit;
+  cratesIO = import ./crates-io.nix { inherit lib buildRustCrate buildRustCrateHelpers; };
```

Comment:
Would it not be better to use `rustPlatform.buildRustPackage` instead?
See [ripgrep](https://github.com/NixOS/nixpkgs/blob/35afcc33be6725503ace682d480f648e4727bd91/pkgs/tools/text/ripgrep/default.nix), [rdedup](https://github.com/NixOS/nixpkgs/blob/84cf00f98031e93f389f1eb93c4a7374a33cc0a9/pkgs/tools/backup/rdedup/default.nix), [alacritty](https://github.com/NixOS/nixpkgs/blob/40fab12bd5dcd1e07813bcd571509697d0ba9ea5/pkgs/applications/misc/alacritty/default.nix).

Reply:
This is debated; see https://github.com/NixOS/nixpkgs/issues/84826

Reply:
Nope, it doesn't work because of some incompatibility with the `Cargo.toml`/`Cargo.lock` files. Also I don't want to use `rustPlatform.buildRustPackage` since it abuses IFD.